app.go

```diff
@@ -191,6 +191,24 @@ func (a *App) useShortOptionHandling() bool {
 func (a *App) Run(arguments []string) (err error) {
 	a.Setup()
 
+	commandNamesWithoutDuplicates := make(map[string]bool)
+	for _, c := range a.Commands {
+		_, cExists := commandNamesWithoutDuplicates[c.Name]
+		if cExists {
+			return fmt.Errorf("list of commands contains duplicate Name: %s", c.Name)
+		}
+		commandNamesWithoutDuplicates[c.Name] = true
+
+		subCommandNamesWithoutDuplicates := make(map[string]bool)
+		for _, s := range c.Subcommands {
+			_, sExists := subCommandNamesWithoutDuplicates[s.Name]
+			if sExists {
+				return fmt.Errorf("list of child commands of %s contains duplicate Name: %s", c.Name, s.Name)
```

Comment:
⭐⭐ _(seriously consider)_

```suggestion
				return fmt.Errorf("Your command %q contains multiple subcommands with the Name %q. Having multiple subcommands with the same name results in ambiguous behavior, so please make sure each subcommand in your command has a unique name.", c.Name, s.Name)
```